airflow/models/connection.py

```diff
@@ -137,7 +137,25 @@ def __init__(  # pylint: disable=too-many-arguments
             self.password = password
             self.schema = schema
             self.port = port
-            self.extra = extra
+            self.extra = self.validate_extra(extra)
+
+    @staticmethod
+    def validate_extra(extra: str):
+        """
+        `extra` parameter is a JSON encoded object. This methods validates that the data
+        adheres to this specification.
+
+        :param extra: The extra section of the .
+        :type extra: str
+
+        :return str
+        """
+        try:
+            json.loads(extra)
```

Comment:
@mik-laj has suggested we should forbid the case where `json.loads(extra)` produces a list or primitive

anyone else agree with that?